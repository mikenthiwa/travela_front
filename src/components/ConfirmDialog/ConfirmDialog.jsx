import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Modal from '../modal/Modal';
import './ConfirmDialog.scss';
import ButtonLoadingIcon from '../Forms/ButtonLoadingIcon';

class ConfirmDialog extends Component {

  renderModal = (id,renderDialogText, modalInvisible, closeDeleteModal,
    buttonSelected,handleApprove, handleReject, handleVerify, documentText, isConfirmDialogLoading,
    customOverlayClass, customButtonClass) => {
    return(
      <Modal
        customModalStyles="delete-comment-modal"
        customOverlayStyle={
          customOverlayClass || `${renderDialogText(buttonSelected)}-modal-overlay`}
        visibility={modalInvisible ? 'invisible' : 'visible'}
        closeDeleteModal={closeDeleteModal}
        title={`${buttonSelected} ${documentText}?`}
        showOverlay={false}
      >
        <p className="approval-comment-modal__text">
          {`Confirm ${renderDialogText(buttonSelected)}`}
        </p>
        <button
          className={`${ customButtonClass || renderDialogText(buttonSelected)}-comment-modal__btn`}
          type="button"
          id={buttonSelected}
          disabled={isConfirmDialogLoading}
          onClick={
            (['Approve', 'Verify', 'approve'].includes(buttonSelected))
              ? () => handleApprove(id) || handleVerify(id) : () => handleReject(id)
          }
        >
          <ButtonLoadingIcon isLoading={isConfirmDialogLoading} buttonText={buttonSelected} />
        </button>
      </Modal>
    );
  };

  renderConfirmModal() {
    const {
      id,
      modalInvisible,
      buttonSelected,
      renderDialogText,
      closeDeleteModal,
      handleApprove,
      handleReject,
      handleVerify,
      documentText,
      isConfirmDialogLoading,
      customOverlayClass,
      customButtonClass
    } = this.props;
    return (
      this.renderModal(id,renderDialogText, modalInvisible, closeDeleteModal,buttonSelected,
        handleApprove,handleReject,handleVerify, documentText, isConfirmDialogLoading,
        customOverlayClass, customButtonClass)
    );
  }

  render() {
    return (
      <Fragment>
        {this.renderConfirmModal()}
      </Fragment>
    );
  }
}

ConfirmDialog.propTypes = {
  id: PropTypes.string,
  modalInvisible: PropTypes.bool,
  buttonSelected: PropTypes.oneOfType([
    PropTypes.bool, PropTypes.string
  ]),
  handleVerify: PropTypes.func,
  renderDialogText: PropTypes.func.isRequired,
  closeDeleteModal: PropTypes.func.isRequired,
  handleApprove: PropTypes.func,
  handleReject: PropTypes.func.isRequired,
  documentText: PropTypes.string,
  isConfirmDialogLoading: PropTypes.bool,
  customOverlayClass: PropTypes.string,
  customButtonClass: PropTypes.string
};

ConfirmDialog.defaultProps = {
  id: '',
  modalInvisible: true,
  buttonSelected: null,
  handleVerify: () => {},
  handleApprove: () => {},
  documentText: null,
  isConfirmDialogLoading: false,
  customOverlayClass: null,
  customButtonClass: null
};

export default ConfirmDialog;
