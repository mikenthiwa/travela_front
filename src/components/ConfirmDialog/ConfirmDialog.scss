@media (min-width: 768px) and (max-width: 769px) {
  .approval-modal-overlay {
    left: 343px !important;
    top: 145px !important;
    height: 120px !important;
  }

  .rejection-modal-overlay {
    left: 444px !important;
    top: 145px !important;
    outline: none !important;
    height: 120px !important;
  }

  .verification-modal-overlay, .cancellation-modal-overlay {
    left: 343px !important;
    top: 145px !important;
    outline: none !important;
    height: 120px !important;
  }
}
@media (min-width: 638px) and (max-width: 767px) {
  .approval-modal-overlay {
    left: 265px !important;
    top: 145px !important;
    height: 120px !important;
  }

  .rejection-modal-overlay {
    left: 366px !important;
    top: 145px !important;
    outline: none !important;
    height: 120px !important;
  }

  .verification-modal-overlay , .cancellation-modal-overlay{
    left: 366px !important;
    top: 145px !important;
    outline: none !important;
    height: 120px !important;
  }
}
@media (max-width: 568px) {
  .approval-modal-overlay {
    left: 30px !important;
    top: 190px !important;
    height: 120px !important;
  }

  .rejection-modal-overlay {
    left: 144px !important;
    top: 190px !important;
    outline: none !important;
    height: 120px !important;
  }

  .verification-modal-overlay , .cancellation-modal-overlay{
    left: 30px !important;
    outline: none !important;
    height: 120px !important;
  }
}
@media (min-width: 561px) and (max-width: 620px) {
  .approval-modal-overlay {
    left: 220px !important;
    top: 140px !important;
    height: 120px !important;
  }

  .rejection-modal-overlay {
    left: 260px !important;
    top: 140px !important;
    outline: none !important;
    height: 120px !important;
  }

  .verification-modal-overlay, .cancellation-modal-overlay{
    left: 260px !important;
    top: 140px !important;
    outline: none !important;
    height: 120px !important;
  }
}
@media (min-width: 496px) and (max-width: 560px) {
  .approval-modal-overlay {
    left: 160px !important;
    top: 140px !important;
  }

  .rejection-modal-overlay {
    left: 260px !important;
    top: 140px !important;
    outline: none !important;
  }

  .verification-modal-overlay , .cancellation-modal-overlay{
    left: 260px !important;
    top: 140px !important;
    outline: none !important;
  }
}
.approval-modal-overlay  {
  position: absolute !important;
  height: 120px;
  width: 243px !important;
  top: 140px;
  left: 477px;
  border: 1px solid #ddd;
  outline: none;
  overflow: hidden;
  box-shadow: 0 2px 10px 0 rgba(0,0,0,0.15);

  .modal {
    top: 0;
  }

  .modal-close {
    margin-right: 26px;
  }

  .modal-title-bar .modal-title-text {
    margin-left: 26px;
  }
}

.modal-title-bar {
  color: #4F4F4F;
  font-family: "DIN Pro";
  font-size: 14px;
  line-height: 18px;


  .modal-title-text {
    color: #4F4F4F;
  }
}

.rejection-modal-overlay {
  position: absolute !important;
  height: 120px;
  width: 243px !important;
  top: 140px;
  left: 577px;
  outline: none;
  border: 1px solid #ddd;
  overflow: hidden;
  box-shadow: 0 2px 10px 0 rgba(0,0,0,0.15);

  .modal {
    top: 0;
  }

  .modal-close {
    margin-right: 26px;
  }

  .modal-title-bar .modal-title-text {
    margin-left: 26px;
  }
}

.verification-modal-overlay {
  position: absolute !important;
  height: 120px;
  width: 270px !important;
  top: 130px;
  left: 440px;
  outline: none;
  overflow: hidden;
  border: 1px solid #ddd;
  box-shadow: 0 2px 10px 0 rgba(0,0,0,0.15);

  .modal {
    top: 0;
  }

  .modal-close {
    margin-right: 26px;
  }

  .modal-title-bar .modal-title-text {
    margin-left: 26px;
  }
}

.trip-modification-modal-overlay{
  position: absolute !important;
  height: 120px;
  width: 270px !important;
  top: 60px;
  left: 100px;
  outline: none;
  overflow: hidden;
  border: 1px solid #ddd;
  box-shadow: 0 2px 10px 0 rgba(0,0,0,0.15);

  .modal {
    top: 0;
  }

  .modal-close {
    margin-right: 26px;
  }

  .modal-title-bar .modal-title-text {
    margin-left: 26px;
  }
}

.approval-comment-modal{
  &__text {
    height: 18px;
    width: 180px;
    color: #999999;
    font-size: 14px;
    line-height: 18px;
    margin-top: 14px;
    margin-left: 26px;
    text-align: left;
  }

  &__btn {
    height: 26px;
    width: 80px;
    border-radius: 1px;
    background-color: rgb(73, 170, 175);
    border: none;
    color: #ffffff;
    margin-right: 28px;
    cursor: pointer;
  }
}

.rejection-comment-modal__btn,
#Reject.trip-modification-Reject-comment-modal__btn{
  height: 26px;
  width: 80px;
  border-radius: 1px;
  background-color: #FF5359;
  border: none;
  color: #ffffff;
  margin-right: 28px;
  cursor: pointer;
}

#verify.request.verification-comment-modal__btn,
.verification-comment-modal__btn ,
#Approve.trip-modification-Approve-comment-modal__btn{
  height: 26px;
  width: 90px;
  border-radius: 1px;
  background-color: #08c75e;
  border: none;
  color: #ffffff;
  margin-right: 28px;
  cursor: pointer;
}
