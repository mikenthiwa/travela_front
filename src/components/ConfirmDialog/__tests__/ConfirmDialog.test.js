import React from 'react';
import { shallow } from 'enzyme';
import ConfirmDialog from '../ConfirmDialog';

const props = {
  id: '1',
  modalInvisible: true,
  buttonSelected: 'Approve',
  renderDialogText: jest.fn(),
  closeDeleteModal: jest.fn(),
  handleApprove: jest.fn(),
  handleReject: jest.fn(),
  documentText: 'Request',
  isConfirmDialogLoading:false
};


describe('ConfirmDialog component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConfirmDialog {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the ConfirmDialog component as expected', () => {
    expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('Modal').length).toBe(1);
    expect(wrapper.length).toBe(1);
  });

  it('should change the visibility of confirmDialog if modalInvisible is false', () => {
    wrapper.setProps({ buttonSelected: 'Approve' });
    wrapper.setProps({ modalInvisible: false });
    expect(wrapper.props().children.props.visibility).toBe('visible');
  });

  it('should change the visibility of confirmDialog if modalInvisible is true', () => {
    wrapper.setProps({ modalInvisible: true });
    expect(wrapper.props().children.props.visibility).toBe('invisible');
  });

  it('should change the modal title based on button selected', () => {
    wrapper.setProps({ buttonSelected: 'Approve' });
    expect(wrapper.props().children.props.title).toBe('Approve Request?');
  });

  it('should change the modal title based on button selected', () => {
    wrapper.setProps({ buttonSelected: 'Reject' });
    expect(wrapper.props().children.props.title).toBe('Reject Request?');
  });

  it('should call handleApprove method when approve button is clicked', () => {
    wrapper.setProps({ buttonSelected: 'Approve' });
    const button = wrapper.find('#Approve');
    button.simulate('click');
    expect(props.handleApprove).toHaveBeenCalledTimes(1);
  });

  it('should call handleApprove method when Reject button is clicked', () => {
    wrapper.setProps({ buttonSelected: 'Reject' });
    const button = wrapper.find('#Reject');
    button.simulate('click');
    expect(props.handleApprove).toHaveBeenCalledTimes(1);
  });

});
