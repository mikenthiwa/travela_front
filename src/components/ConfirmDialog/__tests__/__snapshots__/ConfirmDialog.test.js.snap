// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog component should match snapshot 1`] = `
<Fragment>
  <Modal
    closeDeleteModal={[Function]}
    closeModal={null}
    customModalStyles="delete-comment-modal"
    customOverlayStyle="undefined-modal-overlay"
    modalBar={<div />}
    modalContentId=""
    modalId=""
    requestId=""
    showOverlay={false}
    title="Approve Request?"
    visibility="invisible"
    width=""
  >
    <p
      className="approval-comment-modal__text"
    >
      Confirm undefined
    </p>
    <button
      className="undefined-comment-modal__btn"
      disabled={false}
      id="Approve"
      onClick={[Function]}
      type="button"
    >
      <ButtonLoadingIcon
        buttonText="Approve"
        isLoading={false}
      />
    </button>
  </Modal>
</Fragment>
`;
